### PR TITLE
[ticket/13693] Allow modifying the MySQL author search query

### PR DESCRIPTION
**Pull request by brunoais**
- [ ] based on https://github.com/phpbb/phpbb/pull/3479

https://tracker.phpbb.com/browse/PHPBB3-13693

### CODE REVIEW

**Comment by Nicofuma** [comment]
about this search events I think we should have the same event name for all of them (the same event for mysql, native and postgres) to set the common parameters and a dedicated one just to edit the stuff specific to one of the engine.

**Comment by brunoais** [comment]
That is not possible without meddling with the code in such way that I believe it is better not to have it in a maintenance-version.
I believe I can get the time for that in the summer. There's not much code.
Anyway, doing this the best way requires a decent amount of code revamp.

**Comment by Nicofuma** [comment]
lets do that later so

File: phpBB/phpbb/search/fulltext_mysql.php [hunk 1]
```diff
@@ -620,6 +628,55 @@ public function author_search($type, $firstpost_only, $sort_by_sql, $sort_key, $
...
+		* @var	string	sort_dir			"a" for ASC or "d" dor DESC for the sort order used
+		* @var	string	sql_sort			The result SQL when processing sort_by_sql + sort_key + sort_dir
+		* @var	string	sort_days			Time, in days, that the oldest post showing can have
+		* @var	string	sql_time			The SQL to search on the time specifyed by sort_days
+		* @var	bool	firstpost_only		Wether or not to search only on the first post of the topics
+		* @var	array	ex_fid_ary			Forum ids that must not be searched on
+		* @var	array	sql_fora			SQL query for ex_fid_ary
+		* @var	string	m_approve_fid_sql	WHERE clause condition on post_visibility restrictions
+		* @var	int		start				How many posts to skip in the search results (used for pagination)
+		* @since 3.1.4-RC1
```
Comment (marc1706) [inline phpBB/phpbb/search/fulltext_mysql.php]: 3.1.5-RC1

**Pull request merged**